program/rust/src/deserialize.rs

```diff
@@ -112,3 +117,45 @@ pub fn initialize_pyth_account_checked<'a, T: PythAccount>(
 
     load_account_as_mut::<T>(account)
 }
+
+fn create_pda_account<'a>(
```

Comment:
This way of creating accounts isn’t safe for PDA’s, as a malicious actor can calculate them upfront and deposit a single lamport in each to make the accounts un-useable.

This is because the create_account instruction refuses to create an account with non-zero lamports, which is also being done further down in `create_pda_if_needed`.

Instead, [see how solitaire create’s all accounts, which is also similar to how Anchor works.](https://github.com/wormhole-foundation/wormhole/blob/dev.v2/solana/solitaire/program/src/types/accounts.rs#L170)

Reply:
Thank you sir!